src/freertos_drivers/ti/CC32xxUart.cxx

```diff
@@ -289,14 +297,34 @@ void CC32xxUart::interrupt_handler()
     while (MAP_UARTCharsAvail(base_))
     {
         long data = MAP_UARTCharGetNonBlocking(base_);
-        if (data >= 0 && data <= 0xff)
+        if (nineBit_)
         {
-            unsigned char c = data;
-            if (rxBuf->put(&c, 1) == 0)
+            if (rxBuf->space() < 2)
             {
                 ++overrunCount;
             }
-            rxBuf->signal_condition_from_isr();
+            else
+            {
+                // parity error bit is moved to the ninth bit, then two bytes
+                // are written to the buffer.
+                long bit9 = (data & 0x200) >> 1;
+                data &= 0xff;
+                data |= bit9;
+                rxBuf->put((uint8_t *)&data, 2);
+                rxBuf->signal_condition_from_isr();
```

Comment:
The if/else [overrun] logic for signal_condition_from_isr() is not the same for both 9 and 8 bit data paths. I think technically either is actually fine, but we should be consistent. Pick one and modify the other to match.

Reply:
fixed